### PR TITLE
Updates deprecated JS listeners in useMediaQuery hook [DNM]

### PR DESCRIPTION
**Pull request by pvaliani**
We should confirm that old browsers which we support can tolerate this change before merging.

addListener support:
image

removeListener support:
image

addEventListener support:
https://caniuse.com/addeventlistener
https://caniuse.com/?search=removeEventListener

**Overall change:**
Updates deprecated JS addListener and removeListener methods to new `addEventListener` and `removeEventListener`

https://developer.mozilla.org/en-US/docs/Web/API/MediaQueryList/addListener

https://developer.mozilla.org/en-US/docs/Web/API/MediaQueryList/removeListener

**Code changes:**

Updates deprecated JS addListener and removeListener to new `addEventListener` and `removeEventListener`.

---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] I have added the `cross-team` label to this PR if it requires visibility across World Service teams
- [x] I have assigned this PR to the Simorgh project
- [x] (BBC contributors only) This PR follows the [repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)

**Testing:**

- [ ] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false yarn test:e2e:interactive`)
- [ ] This PR requires manual testing

### CODE REVIEW

**Review by Louis-Matsika — APPROVED** [review]

**Review by ryanmccombe — APPROVED** [review]

File: src/app/hooks/useMediaQuery/index.jsx [hunk 1]
```diff
@@ -6,8 +6,8 @@ const useMediaQuery = (query, handler) => {
 
     handler(mediaQueryList);
 
-    mediaQueryList.addListener(handler);
-    return () => mediaQueryList.removeListener(handler);
+    mediaQueryList.addEventListener(handler);
+    return () => mediaQueryList.removeEventListener(handler);
```
Comment (ryanmccombe) [inline src/app/hooks/useMediaQuery/index.jsx]:
```suggestion
    mediaQueryList.addEventListener('change', handler);
    return () => mediaQueryList.removeEventListener('change', handler);
```

File: src/app/hooks/useMediaQuery/index.jsx [hunk 2]
```diff
@@ -6,8 +6,8 @@ const useMediaQuery = (query, handler) => {
 
     handler(mediaQueryList);
 
-    mediaQueryList.addListener(handler);
-    return () => mediaQueryList.removeListener(handler);
+    mediaQueryList.addEventListener('change', handler);
```
Comment (ryanmccombe) [inline src/app/hooks/useMediaQuery/index.jsx]: Given the existing method seems to be deprecated, but the new method is also not as widely supported (https://caniuse.com/mdn-api_mediaquerylist_eventlistener_objects) I think we should add a fallback here.

```suggestion
  if (mediaQueryList.addEventListener) {
    mediaQueryList.addEventListener('change', handler);
  } else {
    mediaQueryList.addListener(handler);
  }
```

File: src/app/hooks/useMediaQuery/index.jsx [hunk 3]
```diff
@@ -6,8 +6,8 @@ const useMediaQuery = (query, handler) => {
...
+    return () => mediaQueryList.removeEventListener('change', handler);
```
Comment (ryanmccombe) [inline src/app/hooks/useMediaQuery/index.jsx]:
```suggestion
    return () => {
      if (mediaQueryList.removeEventListener) {
        mediaQueryList.removeEventListener('change', handler)
      } else {
        mediaQueryList.removeListener(handler)
      }
    };
```

**Comment by skwong25** [comment]
**1st round of testing complete 09/12/21 with issues across several browsers:**
- Javascript not working on several browsers (no cookie banner / images not loading / dropdown menu not working)
- Page displaying momentarily and then refreshing with a blank page 

Fix implemented 15/12/21

**2nd round of testing complete 15/12/21 on Level A & B browsers, cross-device**
- Previous issues are resolved.
- Some odd issues noted but these are also appearing on live webpages, to be raised separately: 
      1 Javascript broken on Safari 13.x (desktop & mobile) i.e no cookie banner & images not loading, drop-down menu 
     2. Additional non-functional scroll bar is displaying on Firefox 48.x (desktop).

Testing for this change has passed.

**Pull request merged**